os/fs/vfs/fs_select.c

```diff
@@ -266,7 +266,7 @@ int select(int nfds, FAR fd_set *readfds, FAR fd_set *writefds, FAR fd_set *exce
 
 	DEBUGASSERT(ndx == npfds);
 	if (ndx != npfds) {
-		set_errno(EINVAL);
+		set_errno(EBADF);
```

Comment:
How about keeping EINVAL? In posix, a select set errno as EINVAL when the nfds argument is less than 0 or greater than FD_SETSIZE.

Reply:
ndx != npfds doesn't mean that nfds is negative or  exceeds the RLIMIT_NOFILE resource limit (see getrlimit(2)). Hence we don't need to modify it